src/lua/datetime.lua

```diff
@@ -915,6 +900,36 @@ local function datetime_parse_from(str, obj)
     end
 end
 
+local function datetime_cmp(lhs, rhs, is_raising)
```

Comment:
There's another `datetime_parse_full` call on line 897, where you forgot to pass `false` as 4th parameter. Please do so.

Reply:
Fixed:

```diff
@@ -901,7 +902,7 @@ local function datetime_parse_from(str, obj)
 
     if not fmt or fmt == '' or fmt == 'iso8601' or fmt == 'rfc3339' then
         -- Effect of .tz overrides .tzoffset
-        return datetime_parse_full(str, tzname, offset)
+        return datetime_parse_full(str, tzname, offset, false)
     else
         return datetime_parse_format(str, fmt)
     end
```